lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -143,18 +141,13 @@ public function setOwner($entity, array $assoc)
     /**
      * INTERNAL:
      * Gets the collection owner.
-     *
-     * @return object
      */
-    public function getOwner()
+    public function getOwner(): ?object
     {
         return $this->owner;
     }
 
-    /**
-     * @return Mapping\ClassMetadata
-     */
-    public function getTypeClass()
+    public function getTypeClass(): Mapping\ClassMetadata
```

Comment:
This one is dangerous. We don't enforce type in constructor and there also is ClassMetadataInfo